components/ws-manager-mk2/controllers/workspace_controller.go

```diff
@@ -95,7 +96,6 @@ type WorkspaceReconciler struct {
 func (r *WorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, err error) {
 	span, ctx := tracing.FromContext(ctx, "WorkspaceReconciler.Reconcile")
 	defer tracing.FinishSpan(span, &err)
-	log := log.FromContext(ctx)
```

Comment:
with controller-runtime/pkg/log this added the workspace name to the logger, removing this line will cause any log line below to no longer include the workspace name

I think instead we should do something like:
```
log = log.WithFields(log.OWI(...))
```
and then use that `log` instance in the reconciliation logs

Reply:
(I see the OWI is added in most logs below but not all, it probably makes sense to create this `log` instance so we have it by default on any logs in this method. We can then also remove the explicitly set `OWI` fields in the logs below, so a bit more compact)

Reply:
> with controller-runtime/pkg/log this added the workspace name to the logger

💡 TIL, thank you!

> removing this line will cause any log line below to no longer include the workspace name

🤦 doh!

I wonder if there's a way to add the workspace name to logs, but use `common-go/log`? I say this because the syntax for `controller-runtime/pkg/log` is not consistent with `common-go/log`. I'll take a look and report back.

Reply:
OWI is fine as replacement for workspace name, the name is basically the workspace ID

Reply:
Ahh, got it, thank you! I appreciate the clarifying comment.

Reply:
Hmm, didn't work as expected, let me dig a bit deeper.

Reply:
Here we go.

```go
log.Log.Data = log.OWI(workspace.Spec.Ownership.Owner, workspace.Spec.Ownership.WorkspaceID, workspace.Name)
```

And related log entries:
```
time="2024-04-16T14:52:34Z" level=info msg="updating workspace status" instanceId=2410a528-1e96-47c6-9362-24f7e4cfced5 podStatus="&{nil}" userId=foobar workspaceId=cool-workspace workspaceStatus="&{{1 [redacted] [redacted] Stopped [{WorkspaceContainerRunning False 0 2024-04-16 14:52:34.541186632 +0000 UTC m=+32.520011073 Unknown } {Deployed True 0 2024-04-16 14:52:34 +0000 UTC Unknown } {ContentReady True 0 2024-04-16 14:52:34 +0000 UTC InitializationSuccess } {EverReady True 0 2024-04-16 14:52:34 +0000 UTC Unknown }]  <nil> 0xc0007de880 {  } <nil>}}"
```

Reply:
Avoided logging the workspace name in 445e2553befce85fc35fc4e4948feda9aa57b18a. An alternative would be to `scrubber.Default.Value(workspace.Spec.Ownership.WorkspaceID)`, but that seems unnecessary.

Reply:
Hmm, mutating `log.Log` (the application wide context) seems like a bit too much. I'll see about setting up an instance like you mentioned.

Reply:
`AddFields` sounds good, you could then use `Extract` on each log call to use those fields in the log, e.g.:

```
log.Extract(ctx).WithError(err).Error("unable to create startWorkspace context")
```

Reply:
The logs are no longer visible in GCP when I search for them when using log.Extract. 🤔 I think it may be because the middleware isn't setup, and its okay to result in a no-op.

Also, I see 2 years ago Chris mentioned in a TODO to remove common-go/log. So, I'll revert the addition of common-go/log (oops), and add the OWI another way.

Reply:
Closing in favor of https://github.com/gitpod-io/gitpod/pull/19635